test/UnitTest.cs

```diff
@@ -63,6 +63,14 @@ public void TestReadStringFail()
             System.Collections.Generic.IDictionary<string,string> res = client.ResolveEnv();
             Assert.Equal(secretValue, res[envVarName]);
         }
+
+        [Fact]
+        public void TestResolveEnvFail() {
+            var client = new SecretHub.Client();
+            var ex = Assert.Throws<ApplicationException>(() => client.ResolveEnv());

```

Comment:
Am I getting it right that this test succeeds because of an env definition in the `Makefile`? Is it possible to move the env definition into this test? That makes the test more solid.

Reply:
Yes, that is correct. Unfortunately .NET does not offer a nice way of setting OS environment variables. The variables set with `Environment.SetEnvironmentVariable` are only visible in the .NET process (e.g. if you call `Environment.GetEnvironmentVariable`) so the child process in which the client dll is running will not see them.

Reply:
This is something we might actually want to document:
```c#
Environment.SetEnvironmentVariable("test", "secrethub://test/test/test");
client.ExportEnv(client.ResolveEnv());
return Environment.GetEnvironmentVariable("test");
```
will not return a secret value, but "secrethub://test/test/test"

Reply:
Okay, if that does not work for now, that's a bummer.

Documenting this sounds like a good plan. But wouldn't we be able to solve this by fetching the environment in the C# wrapper and passing it to the Go code?

Reply:
That's a great idea! It would most likely solve this issue. It's a bigger change so let's implement it in another PR. I opened a ticket for it.